src/main/java/frc/robot/commands/TurnTowardsHub.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Vision;
+
+public class TurnTowardsHub extends SequentialCommandGroup{
+    DriveTrain m_driveTrain; 
+    Vision m_vision;
+    double m_toleranceDeg, m_stabilizeSecs;
+    static final double kP = 0.01;
+    class TurnTowardsHubLoop extends CommandBase {
+        boolean m_withinTolerance;
+        double m_lastOutsideToleranceTime;
+        public TurnTowardsHubLoop() {
+        }
+
+        // Called when the command is initially scheduled.
+        @Override
+        public void initialize() {
+            m_withinTolerance = false;
+            m_lastOutsideToleranceTime = Timer.getFPGATimestamp();
+        }
+
+        // Called every time the scheduler runs while the command is scheduled.
+        @Override
+        public void execute() {
+            double errorDeg = m_vision.getRobotAngle();
+            m_withinTolerance = Math.abs(errorDeg) < m_toleranceDeg;
+            if(!m_withinTolerance) m_lastOutsideToleranceTime = Timer.getFPGATimestamp();
+            m_driveTrain.drive(0, -errorDeg * kP, false);
+        }
+
+        // Called once the command ends or is interrupted.
+        @Override
+        public void end(boolean interrupted) {
+            m_driveTrain.drive(0, 0, false);
+            m_vision.setMode(Vision.VisionMode.INTAKE);
```

Comment:
You can't do this here. This will turn on the finder and the ShooterCommand won't be able to get the distance.

Reply:
I think the shooter command can use the last vision result (which is cashed)

Reply:
We shouldn't change the VIsionMode here. We should not change the VisionMode until after we shoot, so this call should go in the end() method of the ShooterCommand.

Reply:
I would prefer that ShooterCommand be generic and only deal with shooting balls. But definitely, the resetting of the Vision mode should be done as part of the cleanup for the meta-command which turns and shoots.